tests/integration/test_nan_reduction.py

```diff
@@ -0,0 +1,259 @@
+from math import prod
+
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+
+import cunumeric as num
+
+NAN_FUNCS = ("nanmax", "nanmin", "nanprod", "nansum")
+
+NDIMS = range(LEGATE_MAX_DIM + 1)
+
+
+class TestNanReductions:
+    """
+    These are positive cases compared with numpy
+    """
+
+    @pytest.mark.parametrize("func_name", NAN_FUNCS)
+    @pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+    @pytest.mark.parametrize("keepdims", [True, False])
+    def test_basic(self, func_name, ndim, keepdims):
+        """This test sets an element to NaN and checks if the output
+        from cuNumeric and numpy match."""
+        shape = (5,) * ndim
+        size = prod(shape)
+        in_np = np.random.random(shape)
+
+        # set an element to nan
+        index_nan = np.random.randint(low=0, high=size)
+        index_nan = np.unravel_index(index_nan, shape)
+        in_num = num.array(in_np)
+        in_num[index_nan] = num.nan
+        in_np[index_nan] = np.nan
+
+        func_np = getattr(np, func_name)
+        func_num = getattr(num, func_name)
+
+        out_np = func_np(in_np, keepdims=keepdims)
+        out_num = func_num(in_num, keepdims=keepdims)
+
+        # relax criteria when doing floating ops with nans excluded
+        if func_np.__name__ == "nanprod" or func_np.__name__ == "nansum":
+            assert np.allclose(out_num, out_np, rtol=1e-4)
+        else:
+            assert np.array_equal(out_num, out_np)
```

Comment:
I would personally advise splitting the test, even at the expense of redundant test code (redundancy is OK in test code!), rather that have a conditional like this in a test. These functions probably have different kinds typical failure modes, having tests with only one assertion means those failures will show up with better granularity.

Reply:
Okay. I've split this particular test. So, it looks like the recommended best practice when we have assertions that are dependent on some parameters of the test is to just remove that dependency/parameterization and write multiple tests.

Reply:
There's always exceptions, but generally speaking, yes. My standing advice is to follow the ["Arrange, Act, Assert"](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) pattern